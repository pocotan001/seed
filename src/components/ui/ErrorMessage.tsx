import styled from "styled-components";
import margin, { IMarginProps } from "~/components/styles/extends/margin";
import { Color } from "~/components/styles/theme";

interface IErrorMessageProps
  extends React.HTMLAttributes<HTMLParagraphElement>,
    IMarginProps {
  children: React.ReactNode;
}

const ErrorMessage = styled<IErrorMessageProps, "p">("p").attrs({
  role: "alert"
})`
  color: ${Color.pink800};
  white-space: pre;
  ${margin};
`;

export default ErrorMessage;
