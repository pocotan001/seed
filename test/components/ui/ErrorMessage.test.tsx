import * as React from "react";
import * as renderer from "react-test-renderer";
import ErrorMessage from "~/components/ui/ErrorMessage";

describe("<ErrorMessage>", () => {
  it("should render correctly", () => {
    const tree = renderer.create(<ErrorMessage>alo</ErrorMessage>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
