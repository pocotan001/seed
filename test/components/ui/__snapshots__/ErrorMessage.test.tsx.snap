// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorMessage> should render correctly 1`] = `
<p
  className="sc-bdVaJa oxfiO"
  role="alert"
>
  alo
</p>
`;
